src/setuptools_scm/__main__.py

```diff
@@ -1,13 +1,21 @@
 import sys
 
+from setuptools_scm import _get_version
 from setuptools_scm import get_version
+from setuptools_scm.config import Configuration
 from setuptools_scm.integration import find_files
 
 
 def main() -> None:
-    print("Guessed Version", get_version())
+    files = list(sorted(find_files("."), key=len))
```

Comment:
This operation is a bit expensive (just to find the `pyproject.toml` at the root), however if we want to consider all SCM including 3rd party extensions, it becomes a bit tricky.

Alternatively `file_finder_hg._hg_toplevel` or `file_finder_git._git_toplevel` could be used, however that will not include other SCM that might be supported via extensions...

A natural question would be: is this command required to be very optimised? Can't the user afford a bit of time here just to check how `setuptools_scm` is returning the version? (I imagine this command would just be used for debugging purposes, once the project is packaged, the version is already set in the distribution file...)

Reply:
This could be an alternative implementation:

```python
import os
import sys

from setuptools_scm import _get_version, get_version
from setuptools_scm.config import Configuration
from setuptools_scm.discover import walk_potential_roots
from setuptools_scm.integration import find_files


def main() -> None:
    try:
        pyproject = _find_config(".")
        print(_get_version(Configuration.from_file(pyproject)))
    except (FileNotFoundError, LookupError):
        print(f"Guessed Version {get_version()}")

    if "ls" in sys.argv:
        for fname in find_files("."):
            print(fname)


def _find_config(parent):
    for directory in walk_potential_roots(parent):
        pyproject = os.path.join(directory, "pyproject.toml")
        if os.path.exists(pyproject):
            return pyproject
    raise FileNotFoundError("pyproject.toml does not exist")

```

Reply:
lets add a flag to make config non-optional and a option to tell where pyproject.toml is

additionally a warning can happen if no pyproject.toml was found